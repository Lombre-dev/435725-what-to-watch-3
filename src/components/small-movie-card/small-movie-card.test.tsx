import * as React from 'react';
import {BrowserRouter} from 'react-router-dom';
import * as renderer from 'react-test-renderer';
import SmallMovieCard from './small-movie-card';

const MOVIE = {
  id: 0,
  title: `The Grand Budapest Hotel`,
  genres: [`Drama`],
  year: 2014,
  poster: `/img/the-grand-budapest-hotel-poster.jpg`,
  frames: [`/img/the-grand-budapest-hotel-poster.jpg`],
  ratingScore: 8.9,
  ratingReviewsCount: 240,
  preview: `/samples/sintel_trailer-480p.mp4`,
  src: `/samples/sintel_trailer-480p.mp4`,
  description: `Description`,
  story: `Story`,
  director: `Wes Andreson`,
  duration: 99,
  actors: [
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Some Actor 1`,
    `Some Actor 2`,
  ],
};
const IS_PREVIEW_ACTIVE = false;
const HANDLE_EVENT = () => {
  // eslint-disable-next-line no-console
  console.log(`handleTestEvent`);
};

describe(`<SmallMovieCard />`, () => {

  it(`render should be match markup`, () => {

    const result = renderer
      .create(<BrowserRouter>
        <SmallMovieCard
          movie={MOVIE}
          isPreviewActive={IS_PREVIEW_ACTIVE}
          onHover={HANDLE_EVENT}
          onLeave={HANDLE_EVENT}
        />
      </BrowserRouter>, {
        createNodeMock: () => {
          return {};
        }
      })
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
