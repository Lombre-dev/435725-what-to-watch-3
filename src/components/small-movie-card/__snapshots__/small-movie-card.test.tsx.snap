// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SmallMovieCard /> render should be match markup 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <a
    className="small-movie-card__link"
    href="/films/0"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      The Grand Budapest Hotel
    </h3>
  </a>
</article>
`;
